src/main/java/com/patchworkmc/Patchwork.java

```diff
@@ -53,79 +55,80 @@
 
 	private static byte[] patchworkGreyscaleIcon;
 
-	static {
-		try {
-			patchworkGreyscaleIcon = Files.readAllBytes(new File(Patchwork.class.getResource("/patchwork-icon-greyscale.png").getPath()).toPath());
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
+	private Path inputDir, outputDir, dataDir, tempDir;
+	private List<IMappingProvider> mappingProviders;
+	private NaiveRemapper naiveRemapper;
+	private ManifestRemapper manifestRemapper;
+	private boolean closed = false;
 
+	static {
 		// TODO: With the new logger from application-core, this is
 		// 		 a little problem, since it does not follow the concept of
 		//		 component sub loggers (see Logger#sub)
 		LOGGER = new Logger("Patchwork");
 		LOGGER.setWriter(new StreamWriter(true, System.out, System.err), LogLevel.TRACE);
-	}
 
-	public static void main(String[] args) throws Exception {
-		File current = new File(System.getProperty("user.dir"));
-		Path currentPath = current.toPath();
-		File voldemapTiny = new File(current, "data/mappings/voldemap-" + version + ".tiny");
-		List<TsrgClass<RawMapping>> classes = Tsrg.readMappings(new FileInputStream(new File(current, "data/mappings/voldemap-" + version + ".tsrg")));
-
-		IMappingProvider intermediary = TinyUtils.createTinyMappingProvider(currentPath.resolve("data/mappings/intermediary-" + version + ".tiny"), "official", "intermediary");
-		TsrgMappings mappings = new TsrgMappings(classes, intermediary);
-
-		if (!voldemapTiny.exists()) {
-			TinyWriter tinyWriter = new TinyWriter("official", "srg");
-			mappings.load(tinyWriter);
-			String tiny = tinyWriter.toString();
-			Files.write(voldemapTiny.toPath(), tiny.getBytes(StandardCharsets.UTF_8));
+		try {
+			patchworkGreyscaleIcon = Files.readAllBytes(new File(Patchwork.class.getResource("/patchwork-icon-greyscale.png").getPath()).toPath());
+		} catch (IOException ex) {
+			LOGGER.thrown(LogLevel.FATAL, ex);
 		}
+	}
 
-		File voldemapBridged = new File(current, "data/mappings/voldemap-bridged-" + version + ".tiny");
-		IMappingProvider bridged;
-
-		if (!voldemapBridged.exists()) {
-			LOGGER.trace("Generating bridged (srg -> intermediary) tiny mappings");
-
-			TinyWriter tinyWriter = new TinyWriter("srg", "intermediary");
-			bridged = new BridgedMappings(mappings, intermediary);
-			bridged.load(tinyWriter);
-
-			Files.write(voldemapBridged.toPath(), tinyWriter.toString().getBytes(StandardCharsets.UTF_8));
-		} else {
-			LOGGER.trace("Using cached bridged (srg -> intermediary) tiny mappings");
+	public Patchwork(Path inputDir, Path outputDir, Path dataDir, Path tempDir, IMappingProvider... mappings) {
```

Comment:
I think it makes sense here to split the `dev` mapping providers from the primary mapping provider, maybe take `IMappingProvider primaryMappings, List<IMappingProvider> devMappings`